core/src/main/java/com/microsoft/applicationinsights/internal/config/TelemetryConfigurationFactory.java

```diff
@@ -333,75 +331,57 @@ private boolean setChannel(ChannelXmlElement channelXmlElement, TelemetryConfigu
     private String getConfigurationFile() {
 
         // Trying to load configuration as a resource.
-        ClassLoader classLoader = TelemetryConfigurationFactory.class.getClassLoader();
-        URL resource = classLoader.getResource(fileToParse);
-        if (resource != null) {
-            String configurationFile = resource.getFile();
-            InternalLogger.INSTANCE.logAlways(InternalLogger.LoggingLevel.INFO, "Configuration file has been successfully found in: '%s'", configurationFile);
-            return configurationFile;
-        }
+        String configurationFile = getConfigurationFromDefaultClassLoader();
 
-        InternalLogger.INSTANCE.logAlways(InternalLogger.LoggingLevel.INFO, "Could not find resource named '%s'", fileToParse);
         // If not found as a resource, trying to load from the executing jar directory
-        String configurationFile =  getConfigurationFromLibraryLocation();
-        if (configurationFile != null) {
-            return configurationFile;
+        if (configurationFile == null) {
+            configurationFile =  getConfigurationFromLibraryLocation();
         }
 
-        // If not found as a resource, trying to load from the class path
-        return getConfFromClassPath(classLoader);
-    }
-
-    private String getConfFromClassPath(ClassLoader classLoader) {
-        if (!(classLoader instanceof URLClassLoader)) {
-            return null;
+        // If still not found try to get it from the class path
+        if (configurationFile == null) {
+            configurationFile =  getConfFromClassPath();
         }
 
-        HashSet<String> checkedUrls = new HashSet<String>();
-
-        URL[] urls = ((URLClassLoader)classLoader).getURLs();
-        for (URL url : urls) {
-            String configurationPath = url.getPath();
-            if (configurationPath.endsWith(".jar")) {
-                int index = configurationPath.lastIndexOf('/');
-                if (index != -1) {
-                    configurationPath = configurationPath.substring(0, index + 1);
-                } else {
-                    continue;
-                }
-            } else {
-                if (!configurationPath.endsWith("/")) {
-                    configurationPath += '/';
-                }
-            }
+        if (configurationFile != null) {
+            InternalLogger.INSTANCE.logAlways(InternalLogger.LoggingLevel.INFO, "Configuration file found in: '%s'", configurationFile);
```

Comment:
"Configuration file has been successfully found in:......." this is what is written in the Blob